pranav/cart/main.rb

```diff
@@ -0,0 +1,192 @@
+class Cart
+    def initialize
+      @menu = { 1 => { name: 'aa', price: 10, quantity: 10 }, 2 => { name: 'bb', price: 30, quantity: 10 }, 3 => { name: 'cc', price: 40, quantity: 7 }, 4 => { name: 'dd', price: 70, quantity: 10 }, 5 => { name: 'ee', price: 90, quantity: 10 } }
+  
+      @cart = { 1 => { name: 'bb', price: 30, quantity: 10 }, 2 => { name: 'cc', price: 40, quantity: 7 } }
+  
+      @counter = @menu.size
+      @cart_counter = @cart.size
+    end
+  
+    def cal_gst(amount)
+      gst_amount = 0.0
+      s_gst = 8
+      c_gst = 8
+      gst_amount = (amount / 100) * (c_gst + s_gst)
+      puts "Your GST amount is #{gst_amount}"
+      gst_amount
+    end
+  
+    def cal_discount(amount)
+      puts 'We have 10%, 20% and 30% discount based on your bill'
+      if amount > 300
+        return (amount * 0.3)
+      elsif amount > 200
+        return (amount * 0.2)
+      elsif amount > 100
+        return (amount * 0.1)
+      else
+        return amount * 0.05
+      end
+    end
+  
+    def cal_gross_bill
+      gross_amount = 0.0
+      @cart.each do |_key, value|
+        gross_amount += (value[:price] * value[:quantity]).to_i
+      end
+      puts "Your gross amount is #{gross_amount}"
+      gross_amount
+    end
+  
+    def make_bill
+      gross_amount = cal_gross_bill
+      gst_amount = cal_gst(gross_amount)
+      discount_amount = cal_discount(gross_amount)
+      puts "Your discount amount is: #{discount_amount}"
+      total_amount = gross_amount + gst_amount - discount_amount
+      puts "Your total amount is: #{total_amount}"
+    end
+  
+    def add_to_menu(item_name, item_price, item_quantity)
+      @counter += 1
+      @menu[@counter] = { name: item_name, price: item_price, quantity: item_quantity }
+      puts "#{item_name} added to menu: "
+      puts
+    end
+  
+    def print_store(store)
+      if store == @menu
+        puts "\n\n--------Our Menu--------"
+      else
+        puts "\n\n--------Your Cart--------"
+      end
+      puts 'ID | Name | Price | Quantity'
+      store.each do |key, value|
+        print "#{key}: "
+        value.each do |_key1, value1|
+          print "   #{value1}    "
+        end
+        puts
+      end
+    end
+  
+    def add_to_cart(item_number, item_quantity)
+      if !@menu.key?(item_number)
+        puts 'We dont have any such item'
+        nil
+      elsif item_quantity <= @menu[item_number][:quantity]
```

Comment:
instead of writing large chunk of logic in if elsif conditionals, we can make use of return statements.